src/main/java/org/breedinginsight/services/impl/BreedingMethodServiceImpl.java

```diff
@@ -111,10 +122,64 @@ public void deleteBreedingMethod(UUID programId, UUID breedingMethodId) throws A
     }
 
-    private boolean validateBreedingMethod(ProgramBreedingMethodEntity method) {
-        return StringUtils.isNotBlank(method.getName())
-                && StringUtils.isNotBlank(method.getAbbreviation())
-                && StringUtils.isNotBlank(method.getCategory())
-                && StringUtils.isNotBlank(method.getGeneticDiversity());
+     boolean isMissingRequiredFields(ProgramBreedingMethodEntity method) {
+        return StringUtils.isBlank(method.getName())
+                || StringUtils.isBlank(method.getAbbreviation())
+                || StringUtils.isBlank(method.getCategory())
+                || StringUtils.isBlank(method.getGeneticDiversity());
+    }
+    boolean isDuplicateMethodNameFoundOnList(ProgramBreedingMethodEntity method, List<ProgramBreedingMethodEntity> programBreedingMethodEntityList) {
+        boolean foundDup = false;
+        for (ProgramBreedingMethodEntity testMethod: programBreedingMethodEntityList) {
+            if(isDuplicateName(testMethod, method)){
+                foundDup = true;
+                break;
+            }
+        }
+        return foundDup;
     }
+
+    boolean isDuplicateMethodAbbreviationFoundOnList(ProgramBreedingMethodEntity method, List<ProgramBreedingMethodEntity> programBreedingMethodEntityList) {
+        boolean foundDup = false;
+        for (ProgramBreedingMethodEntity testMethod: programBreedingMethodEntityList) {
+            if(isDuplicateAbbreviation(testMethod, method)){
+                foundDup = true;
+                break;
+            }
+        }
+        return foundDup;
+    }
+
+    boolean isDuplicateName(ProgramBreedingMethodEntity testMethod, ProgramBreedingMethodEntity method) {
+        // SPECIAL CASE: If the two methods are the same method, then they are not duplicates
+        if( (testMethod.getId()!=null) && testMethod.getId().equals(method.getId()) ){
+            return false;
+        }
+
+        boolean isDup = false;
+        if(testMethod.getName()!= null && testMethod.getName().equalsIgnoreCase(method.getName())){
+            isDup = true;
+        }
+        else if(testMethod.getName()==null && method.getName()==null ){
+            isDup = true;
+        }
+        return isDup;
+    }
+
+    boolean isDuplicateAbbreviation(ProgramBreedingMethodEntity testMethod, ProgramBreedingMethodEntity method) {
+        // SPECIAL CASE: If the two methods are the same method, then they are not duplicates
+        if( (testMethod.getId()!=null) && testMethod.getId().equals(method.getId()) ){
+            return false;
+        }
+
+        boolean isDup = false;
+        if(testMethod.getAbbreviation()!= null && testMethod.getAbbreviation().equalsIgnoreCase(method.getAbbreviation())){
+            isDup = true;
+        }
+        else if(testMethod.getAbbreviation()==null && method.getAbbreviation()==null ){
+            isDup = true;
+        }
```

Comment:
Isn't this not needed because a validation error would have already been thrown for the abbreviation not being supplied?